docs/design/rosbag2_record_replay_service.md

```diff
@@ -0,0 +1,131 @@
+
+    This parameter only affects the service event topic. if `-v` or `--verbose` is set, the info command shows the number of requests and responses for each service based on the service event. Otherwise, only show the number of service events. Note that parsing the number of request and response need spent time. The duration of the parsing is related to the number of recorded service events.
+
+
+Without `-v` or `--verbose` parameter, info command shows as below example.
+
+```
+Files:             123_0.mcap
+Bag size:          28.2 KiB
+Storage id:        mcap
+Duration:          15.59s
+Start:             May 19 2023 13:22:25.340 (1684473745.340)
+End:               May 19 2023 13:22:40.400 (1684473760.400)
+Messages:          60
+Topic information: Topic: /chatter | Type: std_msgs/msg/String | Count: 16 | Serialization Format: cdr
+                   Topic: /events/write_split | Type: rosbag2_interfaces/msg/WriteSplitEvent | Count: 0 | Serialization Format: cdr
+                   Topic: /parameter_events | Type: rcl_interfaces/msg/ParameterEvent | Count: 0 | Serialization Format: cdr
+                   Topic: /rosout | Type: rcl_interfaces/msg/Log | Count: 44 | Serialization Format: cdr
+
+>>>> The below is new added items for service event topic <<<<<<<
+Service : XX  <== The number of service
+Service information: Service: /xxx/xxx | Type: xxx/xxx/xxx | Event Count: XX | Serialization Format: XX
+                     Service: /xxx/xxx | Type: xxx/xxx/xxx | Event Count: XX | Serialization Format: XX
+```
+
+With `-v` or `--verbose` parameter, info command shows as below example.
+```
+Files:             123_0.mcap
+Bag size:          28.2 KiB
+Storage id:        mcap
+Duration:          15.59s
+Start:             May 19 2023 13:22:25.340 (1684473745.340)
+End:               May 19 2023 13:22:40.400 (1684473760.400)
+Messages:          60
+Topic information: Topic: /chatter | Type: std_msgs/msg/String | Count: 16 | Serialization Format: cdr
+                   Topic: /events/write_split | Type: rosbag2_interfaces/msg/WriteSplitEvent | Count: 0 | Serialization Format: cdr
+                   Topic: /parameter_events | Type: rcl_interfaces/msg/ParameterEvent | Count: 0 | Serialization Format: cdr
+                   Topic: /rosout | Type: rcl_interfaces/msg/Log | Count: 44 | Serialization Format: cdr
+
+>>>> The below is new added items for service <<<<<<<
+Service : XX  <== The number of service
+Service information: Service: /xxx/xxx | Type: xxx/xxx/xxx | Request Count: XX | Response Count: XX | Serialization Format: XX
+                     Service: /xxx/xxx | Type: xxx/xxx/xxx | Request Count: XX | Response Count: XX | Serialization Format: XX
+```
+
+### Expand the 'play' command
+
+Add 2 parameters.  
+- `--services ServiceName1 [[ServiceName2 ...]`
+
+    Decide which services will be played.
+
+- `--service-request-mode`
+
+    Replay service request with recorded sequence and interval instead of playing service event message.
```

Comment:
I am not sure when non-service-request-mode is required though? do we have any use cases for this mode?

Reply:
I think I misunderstand phase 1 (https://github.com/ros2/rosbag2/pull/1359#issuecomment-1595530464). It should only do recording service event (Not want to play it. Phase 2 includes play. So only play request).  I will update it.